rpc/calibration/R3BRpcCal2Hit.cxx

```diff
@@ -128,99 +125,72 @@ InitStatus R3BRpcCal2Hit::ReInit()
 void R3BRpcCal2Hit::Exec(Option_t* opt)
 {
     Reset();
-
     //loop over strip data
-    Int_t nHits = fRpcCalStripDataCA->GetEntries();
+    Int_t nHits = fRpcCalDataCA->GetEntries();
     UInt_t iDetector = 0;
     double charge_left = -1000;
     double charge_right = -1000;
     double time_left=0;
     double time_right = 0;
     double ichn_right= 0;
     double ichn_left= 0;
-
-    bool valid = false;
+    UInt_t inum;
 
     for (Int_t i = 0; i < nHits; i++)
     {
-
-        auto map1 = (R3BRpcStripCalData*)(fRpcCalStripDataCA->At(i));
-
-        valid =true;
-
-        if(map1->GetTotRight() >=  charge_right){
-            charge_right=map1->GetTotRight();
-            time_right=map1->GetTimeRight();
-            ichn_right = map1->GetChannelId();
-        }
-        if(map1->GetTotLeft() >= charge_left){
-
-            charge_left=map1->GetTotLeft();
-            time_left= map1->GetTimeLeft();
-            ichn_left = map1->GetChannelId();
-        }
-
-
-    }
-
-    if(ichn_left == ichn_right && valid ){
-
-        UInt_t inum = iDetector * 41 + ichn_right -1;
-
-        //double position = (time_left-time_right)/2.;
-
-
-        double position = ((((time_left-time_right)/2. - (fParCont1->GetAt(inum)-400)*40./800.)*800./40.)
-                  /(fParCont2->GetAt(inum) - fParCont1->GetAt(inum)))*1500 - fParCont4->GetAt(inum);
+        auto map1 = (R3BRpcCalData*)(fRpcCalDataCA->At(i));
+	iDetector = map1->GetDetId() ;
+        inum = iDetector * 41 + map1->GetChannelId() -1;
+
+	if( iDetector==0){
+         if(map1->GetTotR_B() >=  charge_right){
+             charge_right=map1->GetTotR_B();
+             time_right=map1->GetTimeR_B();
+             ichn_right = map1->GetChannelId();
+         }
+         if(map1->GetTotL_T() >= charge_left){
+
+             charge_left=map1->GetTotL_T();
+             time_left= map1->GetTimeL_T();
+             ichn_left = map1->GetChannelId();
+         } 
+      	 if(ichn_left == ichn_right){
+
+          double position = ((((time_left-time_right)/2. - (fParCont1->GetAt(inum)-400)*40./800.)*800./40.)
+                  /(fParCont2->GetAt(inum) - fParCont1->GetAt(inum)))*1500;// - fParCont4->GetAt(inum);
     
-        double charge =  (charge_left + charge_right)/2.;
+          double charge =  (charge_left + charge_right)/2.;
 
-        double time = (time_left + time_right)/2. - fParCont3->GetAt(inum);
-
-        AddHitStrip(ichn_right,time,position,charge);
-
-    }
-
-    //loop over Pmt data
-    nHits = fRpcCalPmtDataCA->GetEntries();
-    iDetector = 1;
-    for (Int_t i = 0; i < nHits; i++)
-    {
-        auto map2 = (R3BRpcPmtCalData*)(fRpcCalPmtDataCA->At(i));
+          double time = (time_left + time_right)/2. - fParCont3->GetAt(inum);
+	
+	  auto tof = fTimeStitch->GetTime(time - fR3BEventHeader->GetTStart());
```

Comment:
This should be:
auto tof = fTimeStitch->GetTime(time - fR3BEventHeader->GetTStart(), "trb", "vftx");
because time is calculated by TRB3 and the start time comes from a VFTX module, so they have different ranges.